src/modules/block/index.ts

```diff
@@ -0,0 +1,308 @@
+        });
+      } else {
+        (propsAndStubs as Record<string, any>)[
+          key
+        ] = `<div data-id="${child._id}"></div>`;
+      }
+    });
+
+    const fragment = document.createElement('template');
+    const render = Handlebars.compile(this._template, { noEscape: true });
+    fragment.insertAdjacentHTML('afterbegin', render(propsAndStubs));
+
+    const element = fragment.firstElementChild as HTMLElement;
+
+    Object.values(this.children).forEach((child) => {
+      if (Array.isArray(child)) {
+        child.forEach((subChild) => {
+          if (this._isBlocksObject(subChild)) {
+            Object.keys(subChild).forEach((k) => {
+              const block = (subChild as BlocksObject)[k];
+              const stub = element.querySelector(`[data-id="${block._id}"]`);
+
+              if (!stub || !block.element) {
+                return;
+              }
+
+              stub!.replaceWith(block.element);
+            });
+          } else {
+            const block = subChild as Block;
+            const stub = element.querySelector(`[data-id="${block._id}"]`);
+
+            if (!stub || !block.element) {
+              return;
+            }
+
+            stub!.replaceWith(block.element);
+          }
+        });
+      } else {
+        const stub = element.querySelector(`[data-id="${child._id}"]`);
+
+        if (!stub || !child.element) {
+          return;
+        }
+
+        stub!.replaceWith(child.element);
+      }
+    });
+
+    return element;
+  }
+
+  getContent() {
+    return this.element;
+  }
+
+  _isBlocksObject(obj: any): obj is BlocksObject {
+    let isBlocksObject = true;
+    Object.values(obj).forEach((val) => {
+      if (!(val instanceof Block)) {
+        isBlocksObject = false;
+      }
+    });
+    return isBlocksObject;
+  }
+
+  _getChildren(propsAndChildren: Record<string, any | Block>): {
+    children: Children;
+    props: P;
+  } {
+    const children: Children = {};
+    const props: P = {} as unknown as P;
+
+    Object.entries(propsAndChildren).forEach(([key, value]) => {
+      if (
+        value instanceof Block ||
+        (Array.isArray(value) && value[0] instanceof Block) ||
+        (Array.isArray(value) && value.every((el) => this._isBlocksObject(el)))
+      ) {
+        children[key] = value;
+      } else {
+        (props as Record<string, any>)[key] = value;
+      }
+    });
+
+    return { children, props };
+  }
+
+  _makePropsProxy(props: P): P {
+    const self = this;
+
+    return new Proxy(props as unknown as object, {
+      get(target: Record<string, unknown>, prop: string) {
+        const value = target[prop];
+        return typeof value === 'function' ? value.bind(target) : value;
+      },
+      set(target: Record<string, unknown>, prop: string, val) {
+        const oldProps = { ...target };
+        target[prop] = val;
+
+        self.eventBus().emit(Block.EVENTS.FLOW_CDU, oldProps, target);

```

Comment:
Можно использовать стрелочные функции, чтобы не прибегать к замыканию
```suggestion
      set: (target: Record<string, unknown>, prop: string, val) => {
        const oldProps = { ...target };
        target[prop] = val;

        this.eventBus().emit(Block.EVENTS.FLOW_CDU, oldProps, target);
```